### DISCUSSION TITLE
Demand Tariffs

### DISCUSSION THREAD

**Discussion by neil-ennis** [netzero-labs/netzero#22] [Q&A] [answered]
Is there a way to configure NetZero to avoid grid consumption during the daily demand window (16:00 until 21:00)?

The data feed from my electricity provider shows when the demand window is active.  I wanted to check if I need to do anything manually to ensure I don't draw from the grid during this time, or if the app will handle this automatically.

**Comment by rob0101** [comment]
Tesla Self consumption mode ?

**Reply by neil-ennis** [reply]
Thanks Rob.  Yes, this is feasible, but it means I'd never be able to sell power during the demand window, which is probably the most lucrative time to sell it.

Normally it won't be a problem because demand window prices are usually high.  But sometimes, the prices are low enough that it might be tempting for the battery to buy, and I want to prevent that.  One five minute slip-up during the month could cost me $30 if I don't get it right.

**Reply by rob0101** [reply]
I skimmed the Tesla fleet API documentation the other day and came across "demand charges", so I guess there's a way to teach the powerwall about the demand window.
![Uploading Screenshot 2024-07-10 at 3.58.15 PM.jpg…]()

**Reply by zigam** [reply]
> But sometimes, the prices are low enough that it might be tempting for the battery to buy, and I want to prevent that.

Would turning off grid charging during the demand window help then?

**Reply by neil-ennis** [reply]
Brilliant.  I didn't even think of that.  I think that's a good approach.  The only downside is the possibility of consumption during the demand window when the price is low, but I can manage that.

Thanks for the advice.

**Reply by zigam** [reply]
Good luck! Note that there seems to be an issue with the 24.12.3 firmware, where the grid charging option is ignored. I notified Tesla so hopefully the can fix this soon.

**Reply by neil-ennis** [reply]
@zigam Unfortunately the battery isn't supplying power when the demand window starts because it deems the grid price is low enough, which forces the house to consume from the grid.

I think a solution might be for me to take my existing price feed from Amber, and artificially inflate the General Prices during the demand window, so the battery thinks it's too expensive to buy.

Is it possible to give NetZero a custom price feed instead of the standard Amber feed?.  I could write my own API with its own prices in the same format as Amber.

The long term solution is for NetZero to recognize the demand window, and send inflated Buy prices to the battery during this period.  You wouldn't alter the "Sell" price because this controls whether or not to dispatch.

**Comment by neil-ennis** [comment, marked as answer]
Here's how I resolved this issue with NetZero
1. 15 minutes before demand window starts switch Powerwall to Self Consume, and Disable Grid Charging
2. 15 minutes after demand window ends, switch Powerwall to TBC, and Enable Grid Charging
3. Leave Amber Automation Running.  (Previously I disabled it).

This doesn't seem ideal to me, but the combination of these three things prevents any consumption or grid charging during the demand window.  Since the Amber Automation is running, it will still dispatch during the demand window but will prevent consumption during the demand window.

If the Amber Automation is not enabled, I won't be able to dispatch and run the risk of consuming during the demand window.

The ideal solution would be either:
1. Allow us the ability to supply a custom feed in the same format as Amber.  We could transform it, so that consumption prices during the demand window were artificially high, or 
2. Allow us to specify demand times in the NetZero settings, and the app would send artificially high consumption prices during that period.

I know ideal solutions require work, so I'm happy to run with my ad-hoc solution.

**Reply by zigam** [reply]
Thanks for the analysis! I'll add support for demand tariffs to the roadmap.
